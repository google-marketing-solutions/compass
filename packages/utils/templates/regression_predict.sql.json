{
  "model_path": "project.dataset.model",
  "output_table_path": "project.dataset.scored",
  "features_table_path": "project.dataset.table"
}
