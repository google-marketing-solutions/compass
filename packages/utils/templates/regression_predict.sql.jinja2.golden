# Copyright 2022 Google LLC.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SQL Jinja template to batch score based on pretrained LTV model.
#
# Args:
#  output_table_path: Full path to table in BigQuery to save prediction outputs.
#   Ex: project.dataset.prediction.
#  model_path: Full path to model in BigQuery. Ex: project.dataset.model.
#  feature_columns: List of columns containing features. Ex: ['col1', 'col2'].
#  features_table_path: Full path to table containing scoring datset in
#   BigQuery. Ex: project.dataset.table.


CREATE TABLE `project.dataset.scored`
 AS
SELECT
  *
FROM
  ML.PREDICT(MODEL `project.dataset.model`,
  (
    SELECT *
    FROM
      `project.dataset.table`
  ));
